zhaquirks/develco/power_plug.py

```diff
@@ -0,0 +1,141 @@
+"""Smart Plugs."""
+from zigpy.profiles import zha
+from zigpy.quirks import CustomCluster, CustomDevice
+from zigpy.zcl.clusters.general import (
+    Alarms,
+    Basic,
+    DeviceTemperature,
+    Groups,
+    Identify,
+    OnOff,
+    Ota,
+    Scenes,
+    Time,
+)
+from zigpy.zcl.clusters.homeautomation import ElectricalMeasurement
+from zigpy.zcl.clusters.measurement import OccupancySensing
+from zigpy.zcl.clusters.smartenergy import Metering
+
+from zhaquirks.const import (
+    DEVICE_TYPE,
+    ENDPOINTS,
+    INPUT_CLUSTERS,
+    MODELS_INFO,
+    OUTPUT_CLUSTERS,
+    PROFILE_ID,
+)
+from zhaquirks.develco import DEVELCO
+
+
+class DevelcoDeviceTemperature(CustomCluster, DeviceTemperature):
+    """Device Temperature. Modify divisor."""
+
+    DEV_TEMP_ID = 0x0000
+
+    def _update_attribute(self, attrid, value):
+        if attrid == self.DEV_TEMP_ID and value is not None:
+            value = value * 100
+        super()._update_attribute(attrid, value)
+
+
+class DevelcoElectricalMeasurement(CustomCluster, ElectricalMeasurement):
+    """Electrical measurement. Fixes power factor."""
+
+    RMS_VOLTAGE_ID = 0x0505
+    RMS_CURRENT_ID = 0x0508
+    ACTIVE_POWER_ID = 0x050B
+    POWERFACTOR_ID = 0x0510
+
+    """Use current/voltage reading to update power factor."""
+
+    def _update_attribute(self, attrid, value):
+        super()._update_attribute(attrid, value)
+
+        if attrid == self.ACTIVE_POWER_ID:
+            # Power reading is updated. Update power factor as well.
+            self._updatePF()
+
+    def _updatePF(self):
+        voltage = self._attr_cache.get(self.RMS_VOLTAGE_ID, 0)
+        current = self._attr_cache.get(self.RMS_CURRENT_ID, 0)
+        power = self._attr_cache.get(self.ACTIVE_POWER_ID, 0)
+        pf = 0
+
+        if voltage > 0 and current > 0:
+            voltage = voltage / 100
+            current = current / 1000
+            pf = 100 * power / (voltage * current)
+
+            if pf > 100:
+                pf = 100
+            elif pf < 0:
+                pf = 0
+        self._update_attribute(self.POWERFACTOR_ID, pf)
```

Comment:
Can we leave this manual power factor calculation out for now?
I know ZHA creates a "power factor" sensor by default (even though the device does not support it). This happens for a lot of devices and needs to be fixed in ZHA.
We just don't do this manual calculation for any other devices and I wouldn't do start with it for now.

Reply:
Sure, new pull request created!